routes/applications.js

```diff
@@ -1,39 +1,80 @@
 const express = require("express");
-const { getPool } = require('../db');
+const { getPool } = require("../db");
 const pool = getPool();
 const { verifyRole } = require("../middleware/authorization");
-const { ROLE_NAMES } = require("../utils")
-const { respondWithError, isUniqueConstraintViolation, isNotNullConstraintViolation } = require("./helpers")
+const { ROLE_NAMES, STATUS_NAMES } = require("../utils");
+const {
+  respondWithError,
+  isUniqueConstraintViolation,
+  isNotNullConstraintViolation,
+} = require("./helpers");
 const router = express.Router();
 router.use(express.json());
 
 router.get("/applications", verifyRole(ROLE_NAMES.mentor), async (req, res) => {
-    try {
-        const result = await pool.query("SELECT * FROM applications");
-        res.json({ applications: result.rows});
-    } catch (err) {
-        console.error(err);
-        respondWithError(res);
-    }
+  try {
+    const result = await pool.query("SELECT * FROM applications");
+    res.json({ applications: result.rows });
+  } catch (err) {
+    console.error(err);
+    respondWithError(res);
+  }
 });
 
 router.post("/applications", async (req, res) => {
-    const { name, about, email, video_link, discord_nickname } = req.body;
+  const { name, about, email, video_link, discord_nickname } = req.body;
+  try {
+    const result = await pool.query(
+      "INSERT INTO applications (name, about, email, video_link, discord_nickname) VALUES ($1, $2, $3, $4, $5) RETURNING *",
+      [name, about, email, video_link, discord_nickname]
+    );
+    res.status(201).json(result.rows[0]);
+  } catch (err) {
+    console.error(err);
+    if (isUniqueConstraintViolation(err.code)) {
+      return respondWithError(
+        res,
+        409,
+        "Application with this email already exists"
+      );
+    } else if (isNotNullConstraintViolation(err.code)) {
+      return respondWithError(res, 400, err.message);
+    }
+    respondWithError(res);
+  }
+});
+
+router.put(
+  "/applications/:id",
+  verifyRole(ROLE_NAMES.mentor),
+  async (req, res) => {
+    const { id } = req.params;
+    const { status, comment, user_id } = req.body; // user_id = who approved/denied
+    const today = new Date();
```

Comment:
`modified_at` has default value in the database, I think we don't have to explicitly set it here.
1) it's an overload
2) it might cause problems if EC2 instance is using other timezone than RDS, the now() will differ